src/bin/mission_manager/app.cpp

```diff
@@ -792,7 +801,7 @@ void jaiabot::apps::MissionManager::handle_command(const protobuf::Command& comm
 
     // Make sure the command is not a repeat
     // If it is, then we should not handle the command and exit
-    if (prev_command_time_ == command.time())
+    if (prev_command_time_ >= command.time())
```

Comment:
This probably needs to be added to the comment, but this will occasionally cause issues. 

Comment from you:
"DCCL time codec only sends 24 hours (by default, this can be increased) worth of time to save space, and then realigns the time on the receive side. As you can't always count on causality (due to minor time mismatches), it uses the closest time, which means you get +/-12 hours before it's wrong."

**Example of scenario:**
The customer hub had a date: March 8th
The customer bot had a date: April 14th

* Hub - time does not update with ntp sync before sending command to bot
* Bot - records the prev_command_time
* Hub - time syncs to the correct time during the mission
* Hub - sends command to bot
* Bot - checks the command time it no longer accepts the new commands

Reply:
Alternatively we can add a counter to the hub command that isn't the time stamp and use this for duplicate rejection.

Reply:
The scenario I described occurred on a customer system. It was something to do with the encoding of the time on the hub and then the decoding of the time on the bot. 

I like the counter, the only issue is if the hub is turned off during a mission (battery dies or other issue) and then the bot would discard the commands. 

We might want to keep a history of timestamps that we previously accepted.

The bad ntp sync might be the way to go so the hub cannot send commands until the hub gets a good one or the user inputs a time.

Reply:
I think a circular buffer (of some reasonable size, e.g. 100) of the timestamps of previously accepted commands would be more than sufficient to solve this problem. If the timestamp is in the buffer we reject the command as already seen.

Reply:
Separately I think we should definitely warn on invalid NTP sync as this can invalidate data.

Reply:
Yes, that sounds great! 

And I agree. I will make sure we have a ticket.

Reply:
Added ticket: https://jaia-innovation.atlassian.net/browse/JAIA-1900

Reply:
Added buffer of last 100 timestamps: https://github.com/jaiarobotics/jaiabot/pull/1091/commits/7fd10b792d85566bfd9c00b4b0bb47b705052eff